builtin/game/statbars.lua

```diff
@@ -65,19 +67,26 @@ local function update_builtin_statbars(player)
 		hud.id_healthbar = nil
 	end
 
+	local breath     = player:get_breath()
 	local breath_max = player:get_properties().breath_max
 	if flags.breathbar and enable_damage and not immortal and
-			player:get_breath() < breath_max then
+			breath <= breath_max then
 		local number = 2 * scaleToDefault(player, "breath")
-		if hud.id_breathbar == nil then
- 			local hud_def = table.copy(breath_bar_definition)
+		if not hud.id_breathbar and breath < breath_max then
+			local hud_def = table.copy(breath_bar_definition)
 			hud_def.number = number
 			hud.id_breathbar = player:hud_add(hud_def)
-		else
+		elseif hud.id_breathbar then
 			player:hud_change(hud.id_breathbar, "number", number)
 		end
-	elseif hud.id_breathbar then
-		player:hud_remove(hud.id_breathbar)
+	end
+
+	-- Hide the breath bar if breathbar HUD flag is set to false,
+	-- player is immortal, or breath == breath_max
+	if hud.id_breathbar and (not flags.breathbar or immortal or breath == breath_max) then
+		minetest.after(1, function(breath_bar)
+			player:hud_remove(breath_bar)
```

Comment:
If the player leaves within this second, the reference becomes invalid. Either you'll trigger a warning or crash the server (probably former). The sane solution here would be to check whether the player is still online.

Reply:
Will do

Reply:
Done